pyuvdata/calfits.py

```diff
@@ -0,0 +1,194 @@
+                                      # self.flag_array[:, :, :, :, np.newaxis],
+                                      self.quality_array[:, :, :, :, np.newaxis]],
+                                     axis=-1)
+
+        if self.cal_type == 'unknown':
+            raise ValueError("unknown calibration type. Do not know how to"
+                             "store parameters")
+
+        # header ctypes for NAXIS
+        prihdr['CTYPE5'] = ('ANTS', 'Antenna numbering.')
+        prihdr['CUNIT5'] = 'Integer'
+        prihdr['CRVAL5'] = 0
+        prihdr['CDELT5'] = -1
+
+        prihdr['CTYPE3'] = ('TIME', 'Time axis.')
+        prihdr['CUNIT3'] = ('JD', 'Time in julian date format')
+        prihdr['CRVAL3'] = self.time_array[0]
+        prihdr['CDELT3'] = self.integration_time
+
+        # more checks for polarization. check ordering.
+        prihdr['CTYPE2'] = ('JONES', 'Jones matrix array')
+        prihdr['CUNIT2'] = ('Integer', 'representative integer for polarization.')
+        prihdr['CRVAL2'] = self.jones_array[0]  # always start with first jones.
+        prihdr['CDELT2'] = -1
+
+        prihdu = fits.PrimaryHDU(data=pridata, header=prihdr)
+
+        col1 = fits.Column(name='ANTNAME', format='8A',
+                           array=self.antenna_names)
+        col2 = fits.Column(name='ANTINDEX', format='D',
+                           array=self.antenna_numbers)
+        cols = fits.ColDefs([col1, col2])
+        ant_hdu = fits.BinTableHDU.from_columns(cols)
+
+        sechdr = fits.Header()
+        secdata = self.flag_array.astype(np.int64) # Can't be bool
+        prihdu = fits.PrimaryHDU(data=pridata, header=prihdr)
+        sechdu = fits.ImageHDU(data=secdata, header=sechdr)
+        hdulist = fits.HDUList([prihdu, ant_hdu, sechdu])
+        hdulist.writeto(filename, clobber=clobber)
+
+    def read_calfits(self, filename, run_check=True, run_check_acceptability=True):
+        F = fits.open(filename)
+        data = F[0].data
+        hdr = F[0].header.copy()
+
+        antdata = F[1].data
+        self.antenna_names = map(str, antdata['ANTNAME'])
+        self.antenna_numbers = map(int, antdata['ANTINDEX'])
+
+        self.Nfreqs = hdr['NFREQS']
+        self.Njones = hdr['NJONES']
+        self.Ntimes = hdr['NTIMES']
+        self.channel_width = hdr['CHWIDTH']
+        self.integration_time = hdr['INTTIME']
+        self.telescope_name = hdr['TELESCOP']
+        self.history = str(hdr.get('HISTORY', ''))
+        while 'HISTORY' in hdr.keys():
+            hdr.remove('HISTORY')
+        self.Nspws = hdr['NSPWS']
+        self.Nants_data = hdr['NANTSDAT']
+        self.Nants_telescope = hdr['NANTSTEL']
+        self.gain_convention = hdr['GNCONVEN']
+        self.x_orientation = hdr['XORIENT']
+        self.cal_type = hdr['CALTYPE']
+        try:
+            self.observer = hdr['OBSERVER']
+        except: pass
+        try:
+            self.pipeline = hdr['CALPIPE']
```

Comment:
I don't think we have something called pipeline on the uvcal object anymore, so this should be removed.